ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -1269,6 +1274,36 @@ object Serializers {
 
             MethodDef(flags, name, originalName, args, resultType, body)(
                 optimizerHints, optHash)
+          } else if ( // temp: test backward compat
+              flags.namespace == MemberNamespace.Public &&
+              owner == ObjectClass &&
```

Comment:
Instead of replacing `clone_O` directly, have you considered replacing the implementation `ObjectClone.clone` with `Clone(AsInstanceOf(args(0).ref, cloneableClassType))`? It feels like this is less "invasive".

Reply:
Hum, I had not considered that 🤔 It might be better, indeed. Although it means that the deserializer hack has to know about `java.lang.ObjectClone`, which was as a whole an implementation detail of the previous stdlib; whereas with the current implementation it only needs to know about `java.lang.Object`, which is specced.

I'm not sure which of the invasiveness argument or the has-to-know-implementation-details argument is strongest.

Reply:
But isn't this a good thing in this case? Because in the previous stdlib, `java.lang.Object` was not broken, `java.lang.ObjectClone` was. So we are fixing a thing that was knowingly broken. IMO this has nothing to do with specification but with history.

Reply:
Yes, we can look at it that way. I don't have a strong opinion, so I'll go with yours. ;)

Reply:
So, I tried that, but it kind of blew off a bit, because, although `java.lang.ObjectClone.clone(o)` could be inlined, the `LoadModule(ObjectClone$)` remained (its constructor is not pure) and therefore the code size exploded (4 KB more in fastOpt). Of course it's "solved" with the last commit, but it still would have meant that the new linker would produce significantly worse code for the old stdlib.

I reverted to the solution where I patch `j.l.Object.clone()` directly. But if you want I can try and go one step further, and also patch the constructor to empty it out or something.